src/main/java/com/github/jeffnelson/jackson/patch/IPatchField.java

```diff
@@ -0,0 +1,29 @@
+
+package com.github.jeffnelson.jackson.patch;
+
+import java.util.function.Consumer;
+
+/**
+ * 
+ * @author jeff.nelson
+ * @since 1.0.0
+ *
+ * @param <T>
+ *            the type of value this patch field represents
+ */
+public interface IPatchField<T> {
```

Comment:
I think we should replace `shouldPatch` and `isPresent` with a single method, something like `isFieldPresent` or `isFieldProvided` keeping the should patch logic inside the `patch` method. I get that having these two separate methods might make the behavior more customizable, where someone could implement IPatchField to not always patch when the field is present. However, I think that would be enabling / encouraging developers to deviate from the described behavior in [rfc7396](https://tools.ietf.org/html/rfc7396), which I don't think we should do. Since this is a project to implement a standardized behavior, we should limit developers from deviating from that standard.

Reply:
I hear what you're saying. I was going to remove the `isPresent()` default method.

I do want to point out that developers using this should have access to both the boolean and the `value` via some kind of getter. No one is forcing anyone to use the `patch(Consumer)` convenience method. So really anyone can take the information provided and do whatever they want, so behavior deviating from rcf7396 is already enabled.